otx/cli/builder/builder.py

```diff
@@ -155,25 +156,27 @@ def build_task_config(
         """
 
         # Create OTX-workspace
-        # Check whether the workspace is existed or not
-        workspace_path.mkdir(exist_ok=False)
+        workspace_path.mkdir(parents=True, exist_ok=False)
 
         # Load & Save Model Template
-        otx_registry = OTXRegistry(otx_root).filter(task_type=task_type)
-        if model_type:
-            template_lst = [temp for temp in otx_registry.templates if temp.name.lower() == model_type.lower()]
-            if len(template_lst) == 0:
-                raise ValueError(
-                    f"[otx build] {model_type} is not a type supported by OTX {task_type}."
-                    f"\n[otx build] Please refer to 'otx find --template --task_type {task_type}'"
-                )
-            template = template_lst[0]
-        else:
-            template = otx_registry.get(DEFAULT_MODEL_TEMPLATE_ID[task_type.upper()])
-        template_dir = Path(template.model_template_path).parent
+        if not template:
+            otx_registry = OTXRegistry(otx_root).filter(task_type=task_type)
+            if model_type:
+                template_lst = [temp for temp in otx_registry.templates if temp.name.lower() == model_type.lower()]
+                if len(template_lst) == 0:
+                    raise ValueError(
+                        f"[otx build] {model_type} is not a type supported by OTX {task_type}."
+                        f"\n[otx build] Please refer to 'otx find --template --task_type {task_type}'"
+                    )
+                template = template_lst[0]
+            else:
+                template = otx_registry.get(DEFAULT_MODEL_TEMPLATE_ID[task_type.upper()])
+
+        assert template is not None
+        template_dir: Path = Path(template.model_template_path).parent
 
         # Copy task base configuration file
-        task_configuration_path = template_dir / template.hyper_parameters.base_path
+        task_configuration_path = template_dir / str(template.hyper_parameters.base_path)
```

Comment:
Is this causing a problem now? As far as I know, ModelTemplate's hyper_parameters.base_path is str or os.path.

Reply:
It makes mypy error. Because `template` can be None.

Reply:
Then shouldn't we raise an error when the template is None?

Reply:
> Then shouldn't we raise an error when the template is None?

For usecase scenario, template can't be None at this line. What I meant is just for mypy checking.
Since one of the argument of `build_task_config` is ```template: ModelTemplate = None```,
mypy recognized template can be None. So, I explicitly gave a type.

Do you have other idea to resolve this mypy issue?

Reply:
In fact, in this line, template cannot be None, right?

Reply:
This is caused by `template.hyper_parameters`. The `hyper_parameters.base_path` can be None.
So, we need to define the explicit type (i.e. `str`)

![image](https://user-images.githubusercontent.com/80735344/216218139-1ac5bd2f-ff3e-463b-af6f-d3747807c560.png)

Reply:
Then isn't `str(None)` itself strange too?

Reply:
I don't know what happens in inner part.. However, if I remove the `str()`, I got message below:
![image](https://user-images.githubusercontent.com/80735344/216218721-b129be62-709e-4007-8231-536fd03d8f05.png)

Reply:
Probably because this function can now receive a ModelTemplate.
Then it seems appropriate to check `if base_path is None`.

Reply:
resolved. with checking the base_path